cmd/xl-storage.go

```diff
@@ -212,31 +212,22 @@ func newXLStorage(ep Endpoint) (s *xlStorage, err error) {
 		return nil, err
 	}
 
-	var rootDisk bool
-	if globalIsCICD {
-		rootDisk = true
-	} else {
-		if globalRootDiskThreshold > 0 {
-			// When you do not want rely on automatic verification
-			// of rejecting root disks, we need to add this threshold
-			// to ensure that root disks are ignored properly.
-			info, err := disk.GetInfo(path)
-			if err != nil {
-				return nil, err
-			}
-
-			// treat those disks with size less than or equal to the
-			// threshold as rootDisks.
-			rootDisk = info.Total <= globalRootDiskThreshold
-		} else {
-			// When root disk threshold is not set, we rely
-			// on automatic detection - does not work in
-			// container environments.
-			rootDisk, err = disk.IsRootDisk(path, SlashSeparator)
-			if err != nil {
-				return nil, err
-			}
+	rootDisk, err := disk.IsRootDisk(path, SlashSeparator)
+	if err != nil {
+		return nil, err
+	}
+	if !rootDisk && globalRootDiskThreshold > 0 {
+		// If for some reason we couldn't detect the root disk
+		// use - MINIO_ROOTDISK_THRESHOLD_SIZE to figure out if
+		// this disk is a root disk.
+		info, err := disk.GetInfo(path)
+		if err != nil {
+			return nil, err
 		}
+
+		// treat those disks with size less than or equal to the
+		// threshold as rootDisks.
+		rootDisk = info.Total <= globalRootDiskThreshold
```

Comment:
the problem with this approach is there are 'two' calls wasted when there is globalRootDiskThreshold set - the change that was done previously was to ensure that when `globalRootDiskThreshold` is set we have no reason to validate disk.IsRootDisk() or not. 

We simply have to just rely on either globalRootDiskThreshold or disk.IsRootDisk() - it is not required for disk.IsRootdisk() to be false to check globalRootDiskThreshold